src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -1032,6 +1044,39 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger? errorLogger, Cancella
             return existing.WithAdditionalTreeOptions(builder.ToImmutable());
         }
 
+        private CompilerAnalyzerConfigOptionsProvider GetCompilerAnalyzerConfigOptionsProvider(
+            AnalyzerConfigSet? analyzerConfigSet,
+            ImmutableArray<AdditionalText> additionalTextFiles,
+            DiagnosticBag diagnostics,
+            Compilation compilation,
+            ImmutableArray<AnalyzerConfigOptionsResult> sourceFileAnalyzerConfigOptions)
+        {
+            var analyzerConfigProvider = CompilerAnalyzerConfigOptionsProvider.Empty;
+            if (Arguments.AnalyzerConfigPaths.Length > 0)
+            {
+                Debug.Assert(analyzerConfigSet is object);
+                analyzerConfigProvider = analyzerConfigProvider.WithGlobalOptions(new DictionaryAnalyzerConfigOptions(analyzerConfigSet.GetOptionsForSourcePath(string.Empty).AnalyzerOptions));
```

Comment:
How hot is this code path? IIRC, I have previously seen in some traces lots of string allocations in `GetOptionsForSourcePath` and quite afraid this adds a lot more to the existing allocations 😢

Reply:
I think the call to `GetOptionsForSourcePath` was already being called in this scenario before the PR, so it feels like it won't make that specific aspect worse.

Reply:
Ah, I didn't go through the whole diff so didn't see it. Thanks for clarifying @RikkiGibson!